include/Rotation.h

```diff
@@ -0,0 +1,81 @@
+#ifndef ALE_ROTATION_H
+#define ALE_ROTATION_H
+
+#include <memory>
+#include <vector>
+
+namespace ale {
+
+  enum RotationInterpolation {
+    slerp,
+    nlerp
+  };
+
+  class Rotation {
+    public:
+      Rotation();
+      Rotation(double w, double x, double y, double z);
+      Rotation(const std::vector<double>& matrix);
+      Rotation(const std::vector<double>& angles, const std::vector<int>& axes);
+      Rotation(const std::vector<double>& axis, double theta);
+      ~Rotation();
+
+      // Special member functions
+      Rotation(Rotation && other) noexcept;
+      Rotation& operator=(Rotation && other) noexcept;
```

Comment:
This is a move assignment operator. It's used in compound expressions to prevent creating and deleting multiple objects when executing things like `a + (b * c)`. For rotations, this comes into play when chaining multiple rotations together.